src/category_theory/adjunction/fully_faithful.lean

```diff
@@ -63,11 +63,60 @@ instance counit_is_iso_of_R_fully_faithful [full R] [faithful R] : is_iso (adjun
     simp,
   end⟩⟩⟩
 
--- TODO also prove the converses?
--- def L_full_of_unit_is_iso [is_iso (adjunction.unit h)] : full L := sorry
--- def L_faithful_of_unit_is_iso [is_iso (adjunction.unit h)] : faithful L := sorry
--- def R_full_of_counit_is_iso [is_iso (adjunction.counit h)] : full R := sorry
--- def R_faithful_of_counit_is_iso [is_iso (adjunction.counit h)] : faithful R := sorry
+/-- If the unit of an adjunction is an isomorphism, then its inverse on the image of L is given
+by L whiskered with the counit. -/
+@[simp]
+lemma inv_map_unit {X : C} [is_iso (h.unit.app X)] :
+  inv (L.map (h.unit.app X)) = h.counit.app (L.obj X) :=
+is_iso.inv_eq_of_hom_inv_id (h.left_triangle_components)
+
+/-- If the unit of an adjunction is an isomorphism, then its inverse on the image of L is given
+by L whiskered with the counit. -/
+@[simps]
+noncomputable def whisker_left_L_counit_iso_of_is_iso_unit [is_iso (adjunction.unit h)] :
+  L ⋙ R ⋙ L ≅ L :=
+(functor.associator _ _ _).symm ≪≫
+  iso_whisker_right (as_iso (adjunction.unit h)).symm L ≪≫ functor.left_unitor _
+
+/-- If the counit of an adjunction is an isomorphism, then its inverse on the image of R is given
+by R whiskered with the unit. -/
+@[simp]
+lemma inv_counit_map {X : D} [is_iso (h.counit.app X)] :
+  inv (R.map (h.counit.app X)) = (h.unit.app (R.obj X)) :=
+is_iso.inv_eq_of_inv_hom_id (h.right_triangle_components)
+
+@[simps]
+noncomputable def whisker_left_R_unit_iso_of_is_iso_counit [is_iso (h.counit)] :
+  R ≅ (R ⋙ L ⋙ R) :=
+(functor.left_unitor _).symm ≪≫
+  (iso_whisker_right (as_iso (h.counit)).symm R) ≪≫  (functor.associator _ _ _)
+
+/-- If the unit is an isomorphism, then the left adjoint is full-/
+noncomputable
+def L_full_of_unit_is_iso [is_iso (h.unit)] : full L :=
+{ preimage := λ X Y f, ((h.hom_equiv X (L.obj Y)) f) ≫ inv ((h.unit).app Y) }
+
+/-- If the unit is an isomorphism, then the left adjoint is faithful-/
+lemma L_faithful_of_unit_is_iso [is_iso (adjunction.unit h)] : faithful L :=
+{ map_injective' := λ X Y f g H,
+  begin
+    rw ←(h.hom_equiv X (L.obj Y)).apply_eq_iff_eq at H,
+    simpa using H =≫ inv (h.unit.app Y),
+  end }
+
+/-- If the counit is an isomorphism, then the right adjoint is full-/
+noncomputable
+def R_full_of_counit_is_iso [is_iso (h.counit)] : full R :=
+{ preimage := λ X Y f, ((inv (adjunction.counit h)).app X) ≫ (h.hom_equiv (R.obj X) Y).symm f }
```

Comment:
We have simp-lemmas to write `(inv a).app X` as `inv (a.app X)`, so it's often nicer to use the latter version instead: it means simp has to do a little less work in most cases, and it's "the canonical" way of writing these things. Again I won't insist on this change but it's something you might like to know about!